proxy/http/HttpSM.cc

```diff
@@ -2771,7 +2771,8 @@ HttpSM::tunnel_handler_post(int event, void *data)
   }
 
   if (event != HTTP_TUNNEL_EVENT_DONE) {
-    if ((event == VC_EVENT_WRITE_COMPLETE) || (event == VC_EVENT_EOS)) {
+    if ((event == VC_EVENT_WRITE_COMPLETE) || (event == VC_EVENT_EOS) || (event == VC_EVENT_ERROR) ||
```

Comment:
@shinrich and I discussed this a bit. It's a big ugly to have these two long conditionals that are identical but not obviously so. What I would recommend is adding an inline function like `isTxnTerminalEvent` that covers `EOS`, `ERROR`, `INACTIVITY_TIMEOUT` and `INACTIVITY_TIMEOUT` and have
```
if (event == VC_EVENT_WRITE_COMPLETE || isEventTxTerminal(event)) {
```
which would be easier to understand and easier to see the set of events are the same here and on line 2784. Our opinion is this would be useful in no small number of other places in the code as well and help prevent problems like this in the future.